### PR TITLE
test: Only wait for one operator instance to be ready

### PR DESCRIPTION
**Pull request by jrajahalme**
Waiting for only one Cilium operator instance to be ready allows
testing in single node environments, such as Minikube.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

### CODE REVIEW

**Comment by jrajahalme** [comment]
test-me-please

File: test/k8sT/assertionHelpers.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ func ExpectCiliumNotRunning(vm *helpers.Kubectl) {
 // the error returned by that function is nil.
 func ExpectCiliumOperatorReady(vm *helpers.Kubectl) {
 	By("Waiting for cilium-operator to be ready")
-	err := vm.WaitforPods(helpers.CiliumNamespace, "-l name=cilium-operator", longTimeout)
+	err := vm.WaitForSinglePod(helpers.CiliumNamespace, "-l name=cilium-operator", longTimeout)
```
Comment (aditighag) [inline test/k8sT/assertionHelpers.go]: Shouldn't we check if it's a single node deployment, and call `WaitForSinglePod`/`WaitForPods` accordingly?

**Review by aditighag — DISMISSED** [review]
Thanks for this PR.

File: test/k8sT/assertionHelpers.go (hunk 1 above)
Comment (jrajahalme) [reply test/k8sT/assertionHelpers.go]: Maybe, but isn't the whole point of having two operator instances HA, i.e., having one of them operational should be fine? Unless we explicitly want to test that in a specific scenario both of them come up?

Comment (christarazi) [reply test/k8sT/assertionHelpers.go]: It's possible that we make a change in the future where the second operator instance fails and this wouldn't alert anyone that something unexpected happened. I think I would agree with Aditi.

Comment (nebril) [reply test/k8sT/assertionHelpers.go]: If we make any features that require operator to have two running instances for some reason, it should be tested separately as a separate e2e suite. I think this change is fine as it is.

**Review by nebril — APPROVED** [review]

File: test/k8sT/assertionHelpers.go (hunk 1 above)
Comment (jrajahalme) [reply test/k8sT/assertionHelpers.go]: Changed to retain the old behavior if number of k8s nodes >= 2.

**Comment by jrajahalme** [comment]
test-me-please

**Review by christarazi — APPROVED** [review]

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
known flakes (#14463 and #14125) on runtime tests, no other failures.

**Comment by jrajahalme** [comment]
retest-runtime

**Review by aditighag — APPROVED** [review]

**Pull request merged**